### ISSUE TITLE
Way to go direct to 'Browse Data' (plus some more general questions)

### ISSUE THREAD

**Issue by chrisisbd** [sqlitebrowser/sqlitebrowser#509]
I'd like to be able to go direct into the 'Browse Data' tab from the command line.  I.e. something like "sqlitebrowser --browse boat.db", if this isn't possible now then this is an enhancement request.

Also, is there anywhere to ask this sort of question, I'd prefer a mailing list but I guess a forum would do.  Twitter really doesn't seem a sane way to handle technical[ish] discussions.

**Comment by tonypdmtr** [comment]
I have already suggested that the default tab on entry be the browser as it is the most likely place one wants to be in the majority of cases.  I don't even think an option should be needed.  Maybe the option should be there to go directly to one of the other tabs.

**Comment by chrisisbd** [comment]
That still doesn't help getting to a specific table, though I agree it does seem to be a sensible default.

**Comment by tonypdmtr** [comment]
Right, and your suggestion only specified the db file, not the table.

Maybe you meant something like
sqlitebrowser some.db --browse sometable

Then, I would suggest, the browser again to be the default, and on the command line to simply say:
sqlitebrowser some.db -t sometable

(-t for table)

**Comment by chrisisbd** [comment]
Oops, yes, you're right.  I didn't make it clear that I want to go direct to a specific table.

**Comment by zwrmueller** [comment]
as I wrote at an other issue, with big data files it could be pretty much time consuming going directly to the BrowseData tab. I would opt for an option which tab to open first - or some sort of intelligent behaviour like this:
* if the file holds a single table, go directly to BrowseData tab
* if the file holds more than a single table, go to DataStructure tab

Regards
 Ralf

**Comment by tonypdmtr** [comment]
I agree an option is possibly the best approach.  (Preferably a command-line one, and a short one like -t to avoid typing too much all the time that specifies directly the table to browse.)

As for the 'intelligent' behavior I don't think it will actually work in practice because (for me, at least) most databases have more than one table/view, so I would end up back in the original (current) behavior, which I don't enjoy very much.  Unless, of course, it is in addition to having an option as described above.

So, if there is a single table, yes, go to browse that one table, but if there are more tables, check what if the option says which one, else (no option) go to datastucture tab.

(BTW, why does opening a big table cause a delay?  Is the whole table read in all at once instead of pages as they are displayed -- you know with the LIMIT ... OFFSET clause?)

Thanks.

**Comment by zwrmueller** [comment]
I agree with you, Tony.

Why does open big tables is causing delay? 
For one reason see issue: "Useless time consuming internal SQL-commands at start-up". 

Other reason could be: besides just showing the data via BrowsData-pane, just before running the Limit-SQL-query (internally run, as could be seen in the SQL_LOG-pane) for showing the data --> an other SQL-query for *counting* the (whole) data is run, and *that* query is not limited.

This behaviour is slightly different from running the same Limit-SQL-query manually from the Excecute_SQL-pane: here beforehand the Limit-SQL-query again a SQL-command for *counting* the data is run, but this time including the LIMIT clause.

Maybe this is stuff for another "Useless time consuming internal SQL-command" ...

Regards
 Ralf

**Comment by chrisisbd** [comment]
On Tue, Feb 02, 2016 at 03:43:44PM -0800, tonypdmtr wrote:
>    I agree an option is possibly the best approach. (Preferably a
>    command-line one, and a short one like -t to avoid typing too much all
>    the time that specifies directly the table to browse.)
> 
My requirement is for the ability to create a menu item (or an icon,
whatever) to click on to go straight to the database table I want to
edit/view.  Thus the length of the option on the command line isn't
important really.

It would be useful to have the ability to specify somehow where to go
in the table, especially if it's a big one.  Default to 'last n
records' that fit on the page with 'last' meaning most recent.

-- 
Chris Green

**Comment by chrisisbd** [comment]
Excellent!  I have just built from the latest source and there it is, a -t option to specify the table to browse.  Thank you.  I was actually going to try and do it myself (that's why I got the source) but it's done already.

**Comment by justinclift** [comment]
You're completely welcome to add further features + submit them here for inclusion if you want. :grinning:
